bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/actions/PentairControllerActions.java

```diff
@@ -0,0 +1,307 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pentair.internal.actions;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.pentair.internal.handler.helpers.PentairControllerLightMode;
+import org.openhab.binding.pentair.internal.handler.helpers.PentairControllerSchedule;
+import org.openhab.binding.pentair.internal.handler.helpers.PentairHeatStatus;
+import org.openhab.binding.pentair.internal.parser.PentairStandardPacket;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PentairControllerActions } class to be used as base for all action commands to send on Pentair bus
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class PentairControllerActions extends PentairBaseActions {
+    private final Logger logger = LoggerFactory.getLogger(PentairControllerActions.class);
+
+    public enum ControllerCommand {
+        GET_STATUS(0x02, 0x02),
+        GET_CLOCK_SETTINGS(0xC5, 0x05),
+        SET_CLOCK_SETTINGS(0x85, -1),
+        SET_CIRCUIT_SWITCH(0x86, 0x01),
+        GET_LIGHT_GROUPS(0xE7, 0x27),
+        SET_LIGHT_MODE(0x60, 0x01),
+        GET_VALVES(0xDD, 0x1D),
+        GET_CIRCUIT_NAME_FUNCTION(0xCB, 0x0B),
+        SAVE_SCHEDULE(0x91, 0x01),
+        GET_SCHEDULE(0xD1, 0x11),
+        GET_SW_VERSION(0xFD, 0xFC),
+        DELAY_CANCEL(0x83, 0x01),
+        GET_HEAT_STATUS(0xC8, 0x08),
+        SET_HEAT_STATUS(0x88, 0x01);
+
+        public int send, response;
+
+        ControllerCommand(int send, int response) {
+            this.send = send;
+            this.response = response;
+        }
+    }
+
+    // Byte to use after 0xA5 in communicating to controller. Not sure why this changes,
+    // but it requires to be in sync and up-to-date
+    private int preambleByte = -1;
+
+    public void setPreambleByte(int preambleByte) {
+        this.preambleByte = preambleByte;
+    }
+
+    /**
+     * Method to turn on/off a circuit in response to a command from the framework
+     *
+     * @param circuit circuit number
+     * @param state
+     */
+    public boolean setCircuitSwitch(int circuit, boolean state) {
+        byte[] packet = { (byte) 0xA5, (byte) preambleByte, (byte) id, (byte) 0x00 /* source */,
+                (byte) ControllerCommand.SET_CIRCUIT_SWITCH.send, (byte) 0x02, (byte) circuit,
+                (byte) ((state) ? 1 : 0) };
+
+        logger.debug("setCircuitSwitch: {}, {}", circuit, state);
```

Comment:
Can this message be replaced by using the debugger? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging